Lib/fontTools/ufoLib/__init__.py

```diff
@@ -1439,9 +1439,9 @@ def _getGlyphSetFormatVersion3(self, validateRead, validateWrite, layerName=None
 				# not caching this could be slightly expensive,
 				# but caching it will be cumbersome
 				existing = {d.lower() for d in self.layerContents.values()}
-				if not isinstance(layerName, unicode):
+				if not isinstance(layerName, str):
```

Comment:
you can remove the entire `if not isinstance(layerName, str):` block, since a `layerName` as anything else than `str` should now be unsupported/impossible.